lib/strategy.py

```diff
@@ -0,0 +1,22 @@
+from abc import ABC, abstractmethod
+from enum import Enum
+import pandas as pd
+
+class CalculationResult(Enum):
+    """ """
+    INSUFFICIENT_FUNDS = 1
+    TRANSACTION_REJECTED = 2
+    POSITION_UPDATED = 3
+    SUCCESSFUL_REBALANCE = 4
+
+
+class Calculator(ABC):
+    """ """
```

Comment:
Potential requirements - `validate_trade` and `rebalance_positions`? Nothing needs to be too heavy.